### ISSUE TITLE
A GNN-based Meta-Learning Method for Sparse Portfolio Optimization

### ISSUE THREAD

**Issue by kayuksel** [google/evojax#58]
Hello,

Let me start by saying that I am a fan of your work here. I have recently open-sourced by GNN-based meta-learning method for optimization. I have applied it to the sparse index-tracking problem from real-world (after an initial benchmarking on Schwefel function), and it seems to outperform Fast CMA-ES significantly both in terms of producing robust solutions on the blind test set and also in terms of time (total duration and iterations) and space complexity. I include [the link to my repository](https://github.com/kayuksel/generative-opt/) here, in case you would consider adding the method or the benchmarking problem to your repository. Note: GNN, which learns how to generate populations of solutions at each iteration, is trained using gradients retrieved from the loss function, as opposed to black-box ones.

Sincerely, K

**Comment by lerrytang** [comment]
Hi, thanks for introducing your repo!
I believe these are interesting algos/examples to a wide range of practitioners, and we would love to have your PRs.
As for meta-learning, @LlionJ had this [example](https://github.com/google/evojax/blob/main/examples/notebooks/BraxMetaLearning.ipynb) on locomotion tasks, you may want to have a comparison with his results.

**Comment by kayuksel** [comment]
Dear @lerrytang,

I apologize for the rest response. I must have missed the notice. Thanks for referring to @LlionJ's work, I will check in-detail.

I have made a few updates, meanwhile, modified the code for the previous repo shared to perform, robust optimization (by introducing behavioral diversity), in a noisy reward environment (due to monte-carlo optimization, and action noise/corrupting).

I also created another repo with examples of solving 100k problems (as well as 500k param MovieLens 1M matrix factorization):
https://github.com/kayuksel/genmeta-vs-nevergrad

I also created a crowd and predator-prey simulation environment that may be useful in ejovax. I now work on army simulation.
https://github.com/kayuksel/multi-rl-crowd-sim

Please let me know if you would have any questions. Have a great week.

Sincerely, Kamer

**Comment by kayuksel** [comment]
@LlionJ's weight visualization is similar to my meta-learning method (the heart-beats are exploitation-exploration episodes).
https://www.youtube.com/watch?v=g4kbGvCJfXU
